Sources/AardvarkReveal/ARKRevealArchiveBuilder.h

```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright 2020 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Utility for building a tar archive of a Reveal bundle.
+///
+/// This is not a fully functional tar implementation and is not appropriate for use in general purpose applications.
```

Comment:
I'd really rather this not be part of the public API, but [from what I can tell that's not easily done](https://medium.com/5-minute-break-while-coding/creating-swift-framework-with-private-objective-c-members-the-good-the-bad-and-the-ugly-4d726386644b). Hopefully the class name and headerdoc comment make it clear that this is _not_ appropriate for use as a general purpose archival tool.

One day it may make sense to build this out further and move it into the `Aardvark` module, since it could be useful for other types of attachments that have a similar bundle structure, but there's a lot of limitations of it right now that I'd rather wait until we have a real use case to deal with.

Reply:
Correct, this problem would go away if this were written in Swift. I find it a lot easier to work with things that require specific memory layout (the header struct in particular here) in Objective-C than Swift though. If you have suggestions for how this could be ported to Swift without requiring a bunch of unsafe pointers, I'm definitely open to it.

Reply:
😂 I mean... yeah it's certainly harder to write this kind of code in Swift. I don't have much in the way of suggestions here.